Sources/Build/Sanitizers.swift

```diff
@@ -0,0 +1,100 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2018 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+import Basic
+import Utility
+
+/// A set of enabled runtime sanitizers.
+public struct EnabledSanitizers {
+    public enum Sanitizer: String {
+        case address
+        case thread
+    }
+
+    /// A set of enabled sanitizers.
+    private var sanitizers: Set<Sanitizer> = []
+
+    public init() {}
+    public init(from list: [Sanitizer]) {
+        self.sanitizers = Set(list)
+    }
+
+    /// Return an established short name for a sanitizer, e.g. "asan".
+    private func shortname(_ sanitizer: Sanitizer) -> String {
+        switch sanitizer {
+        case .address: return "asan"
+        case .thread: return "tsan"
+        }
+    }
+
+    public func runtimeEnvironment(toolchain: Toolchain) -> [String: String] {
+      #if os(macOS)
+        return macosRuntimeEnvironment(toolchain: toolchain)
+      #else
+        return Process.env
+      #endif
+    }
+
+    public func macosRuntimeEnvironment(toolchain: Toolchain) -> [String: String] {
+        var env = Process.env
```

Comment:
Instead of sanitizers, I think we should let clients control the env. Sanitizers can return the additional env they need to inject.